### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by ThyWoof**

### CODE REVIEW

File: SolastaCommunityExpansion/Patches/FightingStyleFeats/RulesetCharacterHeroPatcher.cs [hunk 1]
```diff
@@ -14,22 +14,24 @@ internal static void Postfix(RulesetCharacterHero __instance, List<FeatDefinitio
             {
                 foreach (FeatureDefinition featureDefinition in feat.Features)
                 {
-                    if (featureDefinition is FeatureDefinitionProficiency)
+                    if (!(featureDefinition is FeatureDefinitionProficiency featureDefinitionProficiency))
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/FightingStyleFeats/RulesetCharacterHeroPatcher.cs:17]: .OfType<> ?

File: SolastaCommunityExpansion/Patches/ConditionRemovedOnSourceTurnStart/RulesetActorPatcher.cs [hunk 2]
```diff
@@ -12,38 +12,42 @@ internal static class RulesetActor_ProcessConditionsMatchingOccurenceType
...
-                        {
-                            foreach (RulesetCondition rulesetCondition in keyValuePair.Value)
-                            {
-                                if (rulesetCondition.SourceGuid == __instance.Guid && rulesetCondition.ConditionDefinition is IConditionRemovedOnSourceTurnStart)
-                                {
-                                    conditionsToRemove.Add(rulesetCondition);
-                                }
-                            }
-                        }
+            foreach (GameLocationCharacter contender in battleService.Battle.AllContenders)
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/ConditionRemovedOnSourceTurnStart/RulesetActorPatcher.cs:27]: .Where(c => c != null ... etc)?

File: SolastaCommunityExpansion/Patches/ConditionRemovedOnSourceTurnStart/RulesetActorPatcher.cs [hunk 3]
```diff
@@ -12,38 +12,42 @@ internal static class RulesetActor_ProcessConditionsMatchingOccurenceType
...
+            {
+                if (contender == null || !contender.Valid || contender.RulesetActor == null)
+                {
+                    continue;
+                }
 
-                        foreach (RulesetCondition conditionToRemove in conditionsToRemove)
+                var conditionsToRemove = new List<RulesetCondition>();
+
+                foreach (KeyValuePair<string, List<RulesetCondition>> keyValuePair in contender.RulesetActor.ConditionsByCategory)
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/ConditionRemovedOnSourceTurnStart/RulesetActorPatcher.cs:36]: You could so a SelectMany() Linq statement to generate the conditionsToRemove in a single line

File: SolastaCommunityExpansion/Patches/CustomFightingStyle/RulesetCharacterHeroPatcher.cs [hunk 4]
```diff
@@ -12,24 +12,26 @@ internal static void Postfix(RulesetCharacterHero __instance)
         {
             foreach (FightingStyleDefinition fightingStyleDefinition in __instance.TrainedFightingStyles)
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/CustomFightingStyle/RulesetCharacterHeroPatcher.cs:13]: OfType<>() ?

**Review by PhilPJL — DISMISSED** [review]
Just pointing out that I would use Linq, but it's not mandatory

File: SolastaCommunityExpansion/Patches/FightingStyleFeats/RulesetCharacterHeroPatcher.cs (hunk 1 above)
Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/FightingStyleFeats/RulesetCharacterHeroPatcher.cs:17]: If we use OfType here will need to cast featureDefinition later in the code. Wrote this way to remove nesting.

File: SolastaCommunityExpansion/Patches/CustomFightingStyle/RulesetCharacterHeroPatcher.cs (hunk 4 above)
Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/CustomFightingStyle/RulesetCharacterHeroPatcher.cs:13]: OfType<>() on these contexts will force casting later on

**Review by PhilPJL — APPROVED** [review]

**Pull request merged**